### PR TITLE
Fix tests & update to ex_machina

### PR DESCRIPTION
**Pull request by danielsp45**
Some of the tests were failing duo to User fixtures being used in the tests, so I added ex-machina

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -33,31 +33,55 @@ defmodule Atomic.MixProject do
...
     [
-      {:bcrypt_elixir, "~> 3.0"},
       {:phoenix, "~> 1.6.14"},
-      {:phoenix_ecto, "~> 4.4"},
+
+      # core
+      {:esbuild, "~> 0.4", runtime: Mix.env() == :dev},
+      {:phoenix_live_reload, "~> 1.2", only: :dev},
+      {:phoenix_live_view, "~> 0.17.5"},
+      {:quantum, "~> 3.0"},
```
Comment (ruilopesm) [inline mix.exs]: Why is `quantum` needed?

Comment (danielsp45) [reply mix.exs]: I'm not sure, it was already there

File: mix.exs [hunk 2]
```diff
@@ -33,31 +33,55 @@ defmodule Atomic.MixProject do
...
+
+      # database
       {:ecto_sql, "~> 3.6"},
+      {:phoenix_ecto, "~> 4.4"},
       {:postgrex, ">= 0.0.0"},
+
+      # security
+      {:bcrypt_elixir, "~> 3.0"},
+      {:argon2_elixir, "~> 3.0"},
```
Comment (ruilopesm) [inline mix.exs]: We should only use one of these algorithms. I think atomic is using bcrypt, but check it yourself.

File: mix.exs [hunk 3]
```diff
@@ -33,31 +33,55 @@ defmodule Atomic.MixProject do
...
+
+      # Uploads
```
Comment (ruilopesm) [inline mix.exs]:
```suggestion
      # uploads
```

File: mix.exs [hunk 4]
```diff
@@ -33,31 +33,55 @@ defmodule Atomic.MixProject do
   # Type `mix help deps` for examples and options.
   defp deps do
     [
-      {:bcrypt_elixir, "~> 3.0"},
       {:phoenix, "~> 1.6.14"},
-      {:phoenix_ecto, "~> 4.4"},
+
+      # core
+      {:esbuild, "~> 0.4", runtime: Mix.env() == :dev},
```
Comment (ruilopesm) [inline mix.exs]: This should be moved to `# frontend` category.

File: mix.exs [hunk 5]
```diff
@@ -33,31 +33,55 @@ defmodule Atomic.MixProject do
...
-      {:flop, "~> 0.17.0"},
-      {:phoenix_live_dashboard, "~> 0.6"},
-      {:esbuild, "~> 0.4", runtime: Mix.env() == :dev},
+      {:swoosh, "~> 1.5"},
+      {:phoenix_swoosh, "~> 1.0"},
+
+      # frontend
       {:tailwind, "~> 0.1", runtime: Mix.env() == :dev},
+      {:flop, "~> 0.17.0"},
+      {:floki, ">= 0.30.0", only: :test},
```
Comment (ruilopesm) [inline mix.exs]: Why is this needed? `floki` specifically.

File: mix.exs (hunk 1 above)
Comment (ruilopesm) [reply mix.exs]: Quantum is intended for cron jobs. We are planning to implement it in bokken ([#155](https://github.com/coderdojobraga/bokken/pull/155)), for example.
You can remove it for now 🙏🏼

Comment (ruilopesm) [reply mix.exs]: It is being used, my bad. Check this module: `Atomic.Quantum.CertificateDelivery`

**Comment by MarioRodrigues10** [comment]
I think instead of doing a PR fixing user fixtures, you could fix all tests since you're already change a few of them, what you think?

**Comment by ruilopesm** [comment]
@MarioRodrigues10 We should leave tests for new comers to make 👀
It is a very good way to understand how the platform works.

File: test/support/factories/accounts_factory.ex [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+  alias Atomic.Accounts.User
+
+  defmacro __using__(_opts) do
+    quote do
+      @roles ~w(admin staff student)a
+
+      def user_factory do
+        %User{
+          id: Faker.UUID.v4(),
```
Comment (MarioRodrigues10) [inline test/support/factories/accounts_factory.ex]:
```suggestion
```

File: test/atomic_web/controllers/user_auth_test.exs [hunk 7]
```diff
@@ -3,17 +3,16 @@ defmodule AtomicWeb.UserAuthTest do
...
-  import Atomic.AccountsFixtures
 
   @remember_me_cookie "_atomic_web_user_remember_me"
 
   setup %{conn: conn} do
     conn =
       conn
       |> Map.replace!(:secret_key_base, AtomicWeb.Endpoint.config(:secret_key_base))
-      |> init_test_session(%{})
+      |> Phoenix.ConnTest.init_test_session(%{})
```
Comment (MarioRodrigues10) [inline test/atomic_web/controllers/user_auth_test.exs]:
```suggestion
      |> init_test_session(%{})
```

File: test/atomic_web/controllers/page_controller_test.exs [hunk 8]
```diff
@@ -1,8 +1,3 @@
 defmodule AtomicWeb.PageControllerTest do
   use AtomicWeb.ConnCase
-
-  test "GET /", %{conn: conn} do
-    conn = get(conn, "/")
-    assert html_response(conn, 200) =~ "Welcome to Phoenix!"
-  end
```
Comment (MarioRodrigues10) [inline test/atomic_web/controllers/page_controller_test.exs]:
```c
  test "GET /", %{conn: conn} do
    conn = get(conn, "/")
    assert html_response(conn, 200) =~ "Listing Activities"
  end
```

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]
Also there's some inconsistency because you erased some live tests and kept others.

**Review by MarioRodrigues10 — APPROVED** [review]

**Comment by MarioRodrigues10** [comment]
Great work :rocket:

**Pull request merged**